network-libp2p/src/network.rs

```diff
@@ -1296,6 +1300,23 @@ impl Network {
             NetworkAction::StopConnecting => {
                 swarm.behaviour_mut().pool.stop_connecting();
             }
+            NetworkAction::ConnectPeersByServices { services, output } => {
```

Comment:
I think we are missing here an `optimal_num_peers` or something like that to indicate the network how many peers ideally it should connect to